src/shared/loop-util.c

```diff
@@ -399,6 +399,26 @@ static int attach_empty_file(int loop, int nr) {
         return 0;
 }
 
+/* Duplicate a file descriptor without O_DIRECT for unaligned probing */
+static int loop_device_fd_dup_for_dissect(int loop_fd) {
+        int r;
+        int dissect_fd;
```

Comment:
Not a matter, but I'd like to merge these lines, also, `_cleanup_close_` attribute can use for `dissect_fd`.
Moreover, please add an assertion:
```
_cleanup_close_ dissect_fd = -1;
int flags, r;

assert(loop_fd >= 0);
```